04-structs/models/pokedex.go

```diff
@@ -0,0 +1,40 @@
+package models
+
+import "fmt"
+
+type Pokedex struct {
+	Pokemons map[int]*Pokemon
+}
+
+func (pkdx *Pokedex) AddPokemon(id int, pkm *Pokemon) {
+	if pkdx.Pokemons == nil {
+		pkdx.Pokemons = make(map[int]*Pokemon)
+	}
+
+	pkdx.Pokemons[id] = pkm
+}
+
+func (pkdx *Pokedex) GetPokemon(id int) (*Pokemon, bool) {
+	pkm, exists := pkdx.Pokemons[id]
+
+	return pkm, exists
+}
+
+func (pkdx *Pokedex) ListAll() []*Pokemon {
+	pokemons := []*Pokemon{}
+
+	for _, pkm := range pkdx.Pokemons {
+		pokemons = append(pokemons, pkm)
+	}
+
+	return pokemons
+}
+
+func (pkdx Pokedex) String() string {
+	pokemonStrings := ""
+	for _, v := range pkdx.Pokemons {
+		pokemonStrings += v.String() + "\n"
+	}
+
+	return fmt.Sprintf("Pokedex: \n Pokemons: \n %s", pokemonStrings)

```

Comment:
Es mas eficiente y queda mas "clean" usar los String builders para esto:

```suggestion
	    var builder strings.Builder
    for _, v := range pkdx.Pokemons {
        builder.WriteString(v.String())
        builder.WriteString("\n")
    }

    return fmt.Sprintf("Pokedex: \n Pokemons: \n %s", builder.String())
```